libraries/plugins/es_objects/es_objects.cpp

```diff
@@ -0,0 +1,336 @@
+         if(l != nullptr)
+            PrepareLimit(l);
+      }
+      else if(value.is<asset_bitasset_data_object>() && _es_objects_asset_bitasset) {
+         auto obj = db.find_object(value);
+         auto ba = static_cast<const asset_bitasset_data_object*>(obj);
+         if(ba != nullptr)
+            PrepareBitAsset(ba);
+      }
+   }
+}
+
+void es_objects_plugin_impl::PrepareProposal(const proposal_object* proposal_object)
+{
+   proposal_struct prop;
+   prop.expiration_time = proposal_object->expiration_time;
+   prop.review_period_time = proposal_object->review_period_time;
+   prop.proposed_transaction = fc::json::to_string(proposal_object->proposed_transaction);
+   prop.required_owner_approvals = fc::json::to_string(proposal_object->required_owner_approvals);
+   prop.available_owner_approvals = fc::json::to_string(proposal_object->available_owner_approvals);
+   prop.required_active_approvals = fc::json::to_string(proposal_object->required_active_approvals);
+   prop.available_key_approvals = fc::json::to_string(proposal_object->available_key_approvals);
+
+   std::string data = fc::json::to_string(prop);
+   prepare = graphene::utilities::createBulk("bitshares-proposal", data, fc::json::to_string(proposal_object->id), 0);
+   bulk.insert(bulk.end(), prepare.begin(), prepare.end());
+   prepare.clear();
+}
+
+void es_objects_plugin_impl::PrepareAccount(const account_object* account_object)
+{
+   account_struct acct;
+   acct.membership_expiration_date = account_object->membership_expiration_date;
+   acct.registrar = account_object->registrar;
+   acct.referrer = account_object->referrer;
+   acct.lifetime_referrer = account_object->lifetime_referrer;
+   acct.network_fee_percentage = account_object->network_fee_percentage;
+   acct.lifetime_referrer_fee_percentage = account_object->lifetime_referrer_fee_percentage;
+   acct.referrer_rewards_percentage = account_object->referrer_rewards_percentage;
+   acct.name = account_object->name;
+   acct.owner_account_auths = fc::json::to_string(account_object->owner.account_auths);
+   acct.owner_key_auths = fc::json::to_string(account_object->owner.key_auths);
+   acct.owner_address_auths = fc::json::to_string(account_object->owner.address_auths);
+   acct.active_account_auths = fc::json::to_string(account_object->active.account_auths);
+   acct.active_key_auths = fc::json::to_string(account_object->active.key_auths);
+   acct.active_address_auths = fc::json::to_string(account_object->active.address_auths);
+   acct.voting_account = account_object->options.voting_account;
+
+   std::string data = fc::json::to_string(acct);
+   prepare = graphene::utilities::createBulk("bitshares-account", data, fc::json::to_string(account_object->id), 0);
+   bulk.insert(bulk.end(), prepare.begin(), prepare.end());
+   prepare.clear();
+}
+
+void es_objects_plugin_impl::PrepareAsset(const asset_object* asset_object)
+{
+   asset_struct asset;
```

Comment:
I'd recommend not use "asset" as a local variable name, since there is a class or struct has same name. I'm not sure if it will cause trouble.